code/modules/unit_tests/lightzones.dm

```diff
@@ -0,0 +1,16 @@
+/**
+ * Lightzone test
+ *
+ * Checks to see if a controller can find a lightzone in a given hour.
+ */
+/datum/unit_test/lightzones
+
+/datum/unit_test/lightzones/Run()
+	for(var/datum/day_night_controller/iterating_controller as anything in subtypesof(/datum/lightzone))
+		iterating_controller = new
```

Comment:
You should use allocate instead of new so that they qdel at the end

Reply:
done